src/effects/onLoaded/test.js

```diff
@@ -0,0 +1,46 @@
+import Immutable from 'seamless-immutable';
+
+import createAction from '../../creators/createAction';
+import createReducer from '../../creators/createReducer';
+
+import onLoaded from '.';
+
+const initialState = {
+  target: null,
+  targetLoading: true,
+  count: 0
+};
+
+const setUp = {
+  state: null
+};
+
+beforeEach(() => {
+  setUp.state = Immutable(initialState);
+});
+
+describe('onLoaded', () => {
+  it('Sets correctly loading target', () => {
+    const reducer = createReducer(setUp.state, {
+      '@@ACTION/TYPE': onLoaded()
+    });
+    const newState = reducer(setUp.state, createAction('@@ACTION/TYPE', 'target'));
+    expect(newState.targetLoading).toBeFalsy();
+  });
+  it('Does not modify other targets', () => {
+    const reducer = createReducer(setUp.state, {
+      '@@ACTION/TYPE': onLoaded()
+    });
+    const newState = reducer(setUp.state, createAction('@@ACTION/TYPE', 'target'));
+    expect(newState.target).toBeNull();
+  });
+  it('Sets loading conditionally', () => {
+    const reducer = createReducer(setUp.state, {
+      '@@ACTION/TYPE': onLoaded((action, state) => action.payload || state.count > 0)
```

Comment:
I'd remove `state.count > 0` from there. That's not being altered in the test and makes the test more confusing because when reading this expect that state.count will be evaluated